src/lavinmq/amqp/stream/filters/gis.cr

```diff
@@ -0,0 +1,291 @@
+        @points.size.times do |i|
+          pi = @points[i]
+          pj = @points[j]
+          # Check if the edge crosses the ray from point to the right
+          # The first condition ensures pj.lon != pi.lon (no vertical edges)
+          if ((pi.lon > point.lon) != (pj.lon > point.lon)) &&
+             (point.lat < (pj.lat - pi.lat) * (point.lon - pi.lon) / (pj.lon - pi.lon) + pi.lat)
+            inside = !inside
+          end
+          j = i
+        end
+        inside
+      end
+
+      def to_s(io : IO)
+        io << "Polygon(" << @points.size << " points)"
+      end
+    end
+
+    class PolygonFilter
+      include StreamFilter
+
+      getter polygon : Polygon
+
+      def initialize(@polygon : Polygon)
+      end
+
+      def match?(headers : AMQP::Table) : Bool
+        if msg_point = Point.from_headers(headers)
+          return polygon.contains?(msg_point)
+        end
+
+        false
+      end
+
+      def to_s(io : IO)
+        io << "PolygonFilter(" << @polygon << ")"
+      end
+    end
+
+    # Parse radius filter from consumer arguments
+    # Expected format: {"lat": Float64, "lon": Float64, "radius_km": Float64}
+    def self.parse_radius_filter(value : ::AMQ::Protocol::Field) : RadiusFilter?
+      return nil unless value.is_a?(::AMQ::Protocol::Table)
+
+      lat = value["lat"]?
+      lon = value["lon"]?
+      radius = value["radius_km"]?
+      return nil unless lat && lon && radius
+
+      lat_f = to_float64?(lat)
+      lon_f = to_float64?(lon)
+      radius_f = to_float64?(radius)
+      return nil unless lat_f && lon_f && radius_f
+      return nil if radius_f <= 0
+
+      RadiusFilter.new(Point.new(lat_f, lon_f), radius_f)
+    rescue ArgumentError
+      nil
+    end
+
+    # Parse bounding box filter from consumer arguments
+    # Expected format: {"min_lat": Float64, "max_lat": Float64, "min_lon": Float64, "max_lon": Float64}
+    def self.parse_bbox_filter(value : ::AMQ::Protocol::Field) : BoundingBoxFilter?
+      return nil unless value.is_a?(::AMQ::Protocol::Table)
+
+      min_lat = to_float64?(value["min_lat"]?)
+      max_lat = to_float64?(value["max_lat"]?)
+      min_lon = to_float64?(value["min_lon"]?)
+      max_lon = to_float64?(value["max_lon"]?)
+      return nil unless min_lat && max_lat && min_lon && max_lon
+
+      bbox = BoundingBox.new(min_lat, max_lat, min_lon, max_lon)
+      BoundingBoxFilter.new(bbox)
+    rescue ArgumentError
+      nil
+    end
+
+    # Parse polygon filter from consumer arguments
+    # Expected format: {"points": [[lat1, lon1], [lat2, lon2], ...]}
+    def self.parse_polygon_filter(value : ::AMQ::Protocol::Field) : PolygonFilter?
+      return nil unless value.is_a?(::AMQ::Protocol::Table)
+
+      points_field = value["points"]?
+      return nil unless points_field.is_a?(Array)
+
+      points = [] of Point
+      points_field.each do |point_field|
+        next unless point_field.is_a?(Array)
+        next unless point_field.size == 2
```

Comment:
Here an invalid value is "ignored", but in other places we fail (like below if lat or lon is nil). Shouldn't we fail here too?